pkg/receiver/gitproviderreceiver/internal/scraper/githubscraper/github_scraper.go

```diff
@@ -62,58 +62,88 @@ func newGitHubScraper(
+	ghs.logger.Sugar().Debugf("pull request pages: %v for repo %v", prPages, repoName)
+	return prPages, err
+}
+
+func getPrData(
+	ghs *githubScraper,
+	ctx context.Context,
+	client graphql.Client,
+	prPages int,
+	repoName string,
+	owner string,
+) ([]PullRequestNode, error) {
+	var prCursor *string
+	var pullRequests []PullRequestNode
+
+	for i := 0; i < prPages; i++ {
+		pr, err := getPullRequestData(ctx, client, repoName, ghs.cfg.GitHubOrg, 100, prCursor)
+		if err != nil {
+			return nil, err
+		}
+
+		pullRequests = append(pullRequests, pr.Repository.PullRequests.Nodes...)
+		prCursor = &pr.Repository.PullRequests.PageInfo.EndCursor
+
+		if !pr.Repository.PullRequests.PageInfo.HasNextPage {
+			break
+		}
+	}
+	return pullRequests, nil
+}
+func getPullRequests(
+	ghs *githubScraper,
+	ctx context.Context,
+	client graphql.Client,
+	repos []SearchNodeRepository,
 	now pcommon.Timestamp,
 	pullRequestCh chan []PullRequestNode,
 	waitGroup *sync.WaitGroup,
 ) {
 	defer waitGroup.Done()
 
 	for _, repo := range repos {
-		var repoName string
-
-		if n, ok := repo.(*SearchNodeRepository); ok {
-			repoName = n.Name
-		}
+		var repoName string = repo.Name
 
 		//var defaultBranch string
-		var prCursor *string
-		var pullRequests []PullRequestNode
 
-		prOpenCount, err := getPullRequestCount(ctx, client, repoName, ghs.cfg.GitHubOrg, []PullRequestState{PullRequestStateOpen})
+		prPages, err := getTotalPrPages(ghs, ctx, client, repoName, ghs.cfg.GitHubOrg, now)
 		if err != nil {
-			ghs.logger.Sugar().Errorf("error getting open pull request count", zap.Error(err))
+			ghs.logger.Sugar().Errorf("error getting total pr pages", zap.Error(err))
 		}
-		ghs.logger.Sugar().Debugf("open pull request count: %v for repo %v", prOpenCount, repoName)
-		ghs.mb.RecordGitRepositoryPullRequestCountDataPoint(now, int64(prOpenCount.Repository.PullRequests.TotalCount), repoName)
-
-		prMergedCount, err := getPullRequestCount(ctx, client, repoName, ghs.cfg.GitHubOrg, []PullRequestState{PullRequestStateMerged})
+		pullRequests, err := getPrData(ghs, ctx, client, prPages, repoName, ghs.cfg.GitHubOrg)
 		if err != nil {
-			ghs.logger.Sugar().Errorf("error getting merged pull request count", zap.Error(err))
-		}
-
-		totalPrCount := add(prOpenCount.Repository.PullRequests.TotalCount, prMergedCount.Repository.PullRequests.TotalCount)
-		prPages := getNumPages(float64(100), float64(totalPrCount))
-		ghs.logger.Sugar().Debugf("pull request pages: %v for repo %v", prPages, repoName)
-
-		for i := 0; i < prPages; i++ {
-			pr, err := getPullRequestData(ctx, client, repoName, ghs.cfg.GitHubOrg, 100, prCursor)
-			if err != nil {
-				ghs.logger.Sugar().Errorf("error getting pull request data", zap.Error(err))
-			}
-
-			pullRequests = append(pullRequests, pr.Repository.PullRequests.Nodes...)
-
-			prCursor = &pr.Repository.PullRequests.PageInfo.EndCursor
+			ghs.logger.Sugar().Errorf("error getting pr data", zap.Error(err))
 		}
 		pullRequestCh <- pullRequests
```

Comment:
`getPullRequests` by name should return pull requests as a value, not a channel.

Reply:
Wouldn't this defeat the point of the go routines? Or do you mean keep the function basic but then just add that to the channel in the go routine outside the function?

Reply:
I think we talked about this in the tech sync today